src/lib/compileAquaAndWatch.ts

```diff
@@ -0,0 +1,198 @@
+  imports: GatherImportsResult;
+  names?: string | undefined;
+  dry?: boolean | undefined;
+  watch?: boolean | undefined;
+};
+
+export async function compileAquaFromFluenceConfig({
+  names: namesFlag,
+  imports,
+  watch = false,
+  dry = false,
+  fluenceConfig,
+}: CompileAquaFromFluenceConfigArgs) {
+  if (!hasAquaToCompile(fluenceConfig)) {
+    return;
+  }
+
+  const names =
+    namesFlag === undefined
+      ? Object.keys(fluenceConfig[COMPILE_AQUA_PROPERTY_NAME])
+      : commaSepStrToArr(namesFlag);
+
+  const [invalidNames, validNames] = splitErrorsAndResults(names, (name) => {
+    if (name in fluenceConfig[COMPILE_AQUA_PROPERTY_NAME]) {
+      return {
+        result: name,
+      };
+    }
+
+    return {
+      error: name,
+    };
+  });
+
+  if (invalidNames.length > 0) {
+    commandObj.error(
+      `${color.yellow(
+        invalidNames
+          .map((name) => {
+            return color.yellow(name);
+          })
+          .join(", "),
+      )} are missing from '${COMPILE_AQUA_PROPERTY_NAME}' property of ${fluenceConfig.$getPath()}`,
+    );
+  }
+
+  const results = await Promise.allSettled(
+    Object.entries(fluenceConfig[COMPILE_AQUA_PROPERTY_NAME])
+      .filter(([name]) => {
+        return validNames.includes(name);
+      })
+      .map(async ([name, aquaConfig]) => {
+        return compileAquaAndWatch(
+          {
+            ...resolveAquaConfig(aquaConfig, imports),
+            outputPath: aquaConfig.output,
+            watch,
+            dry,
+          },
+          name,
+        );
+      }),
+  );
+
+  if (watch) {
+    return;
+  }
+
+  const compilationErrors = results.filter(
+    (res): res is { status: "rejected"; reason: unknown } => {
+      return res.status === "rejected";
+    },
+  );
+
+  if (compilationErrors.length > 0) {
+    commandObj.error(
+      compilationErrors
+        .map((error) => {
+          return stringifyUnknown(error.reason);
+        })
+        .join("\n"),
+    );
+  }
+
+  await exitCli();
+}
+
+export function hasAquaToCompile(
+  maybeFluenceConfig: FluenceConfig | null,
+): maybeFluenceConfig is FluenceConfig & {
+  [COMPILE_AQUA_PROPERTY_NAME]: CompileAqua;
+} {
+  return (
+    maybeFluenceConfig !== null &&
+    COMPILE_AQUA_PROPERTY_NAME in maybeFluenceConfig &&
+    Object.keys(maybeFluenceConfig[COMPILE_AQUA_PROPERTY_NAME]).length !== 0
+  );
+}
+
+export async function compileAquaAndWatch(
+  compileArgs: Required<CompileToFilesArgs> & {
+    watch: boolean;
+  },
+  name?: string,
+) {
+  function compileSuccessLog() {
+    const aquaConfigName = name === undefined ? "" : ` ${color.green(name)}`;
+
+    const from = ` from ${color.yellow(compileArgs.filePath)}`;
+
+    const to =
+      compileArgs.outputPath === undefined || compileArgs.dry
+        ? ""
+        : ` to ${color.yellow(compileArgs.outputPath)}`;
+
+    commandObj.logToStderr(
+      `Successfully compiled${aquaConfigName}${from}${to}`,
+    );
+  }
```

Comment:
```suggestion
  function compileSuccessLog() {
    const aquaConfigName = name === undefined ? "" : `${color.green(name)}`;

    const from = `from ${color.yellow(compileArgs.filePath)}`;

    const to =
      compileArgs.outputPath === undefined || compileArgs.dry
        ? ""
        : `to ${color.yellow(compileArgs.outputPath)}`;

    commandObj.logToStderr(
      `Successfully compiled ${aquaConfigName} ${from} ${to}`,
    );
  }
```

Optional suggestion.